Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp

```diff
@@ -3240,6 +3241,39 @@ CSSRule* Parser::convert_to_rule(NonnullRefPtr<Rule> rule)
 
             return CSSKeyframesRule::create(m_context.realm(), name, move(keyframes)).release_value_but_fixme_should_propagate_errors();
         }
+        if (rule->at_rule_name().equals_ignoring_ascii_case("namespace"sv)) {
+            // https://drafts.csswg.org/css-namespaces/#syntax
+            auto token_stream = TokenStream { rule->prelude() };
+            dbgln("Parsing namespace");
+            auto transaction = token_stream.begin_transaction();
+            token_stream.skip_whitespace();
+
+            auto token = token_stream.peek_token();
+            Optional<StringView> prefix = {};
+            if (token.is(Token::Type::Ident)) {
+                prefix = token.token().ident();
+                token_stream.next_token();
+                token_stream.skip_whitespace();
+                token = token_stream.peek_token();
+            }
+
+            DeprecatedString namespace_uri;
+            if (token.is(Token::Type::String)) {
+                namespace_uri = token.token().string();
+            } else if (token.is_function()) {
+                Optional<AK::URL> url = parse_url_function(token, AllowedDataUrlType::None);
+                if (url.has_value()) {
+                    namespace_uri = url.value().to_deprecated_string();
+                } else {
+                    dbgln_if(CSS_PARSER_DEBUG, "CSSParser: @namespace rule invalid; discarding.");
+                    return {};
+                }
+            } else {
+                dbgln_if(CSS_PARSER_DEBUG, "CSSParser: @namespace rule invalid; discarding.");
+                return {};
+            }
+            return CSSNamespaceRule::create(m_context.realm(), prefix, namespace_uri).release_value_but_fixme_should_propagate_errors();
```

Comment:
Before this line, make sure the syntax is valid, by skipping whitespace and then checking that `token_stream.has_next_token()` is false.